com.unity.uiwidgets/Runtime/foundation/debug.cs

```diff
@@ -57,20 +57,23 @@ public static class D {
 
         [Conditional("UNITY_ASSERTIONS")]
         public static void assert(Func<bool> result, Func<string> message = null) {
-            if (debugLog && !result()) {
+            if ( !result() && !_disableDebugLog) {
```

Comment:
change it to if (!_disableDebugLog && !result()).

if we put result() before, the callback will always be called first regardless of the value of _disableDebugLog